### PR TITLE
feat(rl): Add SFTTrainer for supervised fine-tuning

### LINKED ISSUE #752: SFT Support similar to RLTrainer
Hi Folks, 

A minimal SFTTrainer similar to RLTrainer in verifiers would be amazing. 
Any chance to get it?

### PR DESCRIPTION
**Pull request by gspeter-max**
## Summary

Implements #752: Adds `SFTTrainer` and `SFTConfig` classes to provide a consistent API for supervised fine-tuning, matching the pattern of the existing `RLTrainer`.

### Core Implementation
- **SFTConfig**: Configuration class extending `TrainingArguments` with 40 fields for model loading, LoRA, batch parameters, training parameters, and optional vLLM integration
- **SFTTrainer**: Training class extending `Trainer` with simple cross-entropy loss (no PPO), optional vLLM sampling, and same logging/metrics patterns as `RLTrainer`

### Key Features
- ✅ Consistent API with RLTrainer (same config structure, same initialization)
- ✅ Reuses existing utilities (model loading, logging, LoRA setup)
- ✅ Optional vLLM integration for sample monitoring (not required)
- ✅ Enables easy SFT → RL workflows
- ✅ CLI interface: `vf-sft @ path/to/config.toml`

### Files Modified
- `packages/verifiers-rl/verifiers_rl/rl/trainer/config.py` - Added SFTConfig
- `packages/verifiers-rl/verifiers_rl/rl/trainer/trainer.py` - Added SFTTrainer
- `packages/verifiers-rl/verifiers_rl/rl/trainer/__init__.py` - Added exports
- `packages/verifiers-rl/verifiers_rl/__init__.py` - Added exports
- `verifiers/__init__.py` - Added lazy imports
- `packages/verifiers-rl/verifiers_rl/scripts/sft.py` - Created SFT script
- `packages/verifiers-rl/pyproject.toml` - Added vf-sft entry point

## Usage Example

```python
import verifiers as vf
from datasets import load_dataset

# Load dataset
dataset = load_dataset("willcb/V3-wordle", split="train")

# Create config
config = vf.SFTConfig(
    run_name="wordle-sft",
    max_steps=500,
    learning_rate=2e-5,
    batch_size=512,
    micro_batch_size=8,
)

# Create trainer
trainer = vf.SFTTrainer(
    model="Qwen/Qwen3-4B-Instruct",
    train_dataset=dataset,
    args=config,
)

# Train
trainer.train()
```

Or via CLI:
```bash
vf-sft @ path/to/config.toml
```

Example TOML config:
```toml
model = "Qwen/Qwen3-4B-Instruct"
dataset = "willcb/V3-wordle"

[sft]
run_name = "wordle-sft"
max_steps = 500
learning_rate = 2e-5
batch_size = 512
micro_batch_size = 8
use_lora = true
lora_rank = 8
```

## Design Philosophy

- **Similar Structure**: SFTTrainer follows the same class structure as RLTrainer
- **Simpler Internals**: SFT is inherently simpler (no rollouts, no advantages, no orchestrator)
- **Optional vLLM**: For monitoring sample quality during training (not required)
- **Reuses Existing Utilities**: Leverages all existing model loading, logging, and utilities

Closes #752

---

> [!NOTE]
> **Medium Risk**
> Introduces new training-path code and a new CLI entrypoint, plus touches shared config/LoRA initialization; issues would mainly affect training runs and logging rather than core runtime behavior.
> 
> **Overview**
> Adds **supervised fine-tuning support** alongside the existing RL trainer by introducing `SFTConfig` (a `TrainingArguments`-based config that auto-computes `gradient_accumulation_steps` from `batch_size`/`micro_batch_size` and reuses the LoRA setup) and `SFTTrainer` (a `Trainer` subclass using standard cross-entropy loss, with optional vLLM-based sample generation and logging).
> 
> Exposes the new APIs via `verifiers_rl` and `verifiers` lazy imports, adds a new `vf-sft` TOML-driven CLI script, and updates documentation (`docs/reference.md`, `docs/training.md`, and the training skill guide) with usage examples and an SFT → RL workflow.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2475bf7c9809f3c11d0b4847822aa8f0b814f309. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by xeophon** [comment]
Training models is part of prime-rl, verifiers-rl was mostly meant for illustrative purposes. Thus, I close this PR
